### PR TITLE
Fix generation of auth json map when using a trust id

### PR DESCRIPTION
**Pull request by MatMaul**
gofmt the code in a separate commit.

### CODE REVIEW

**Comment by jrperritt** [comment]
@codevulture thoughts on this?

File: openstack/identity/v3/extensions/trust/request.go [hunk 1]
```diff
@@ -7,34 +7,35 @@ import (
...
+		authMap["scope"] = map[string]interface{}{
+			"OS-TRUST:trust": map[string]interface{}{
+				"id": ao.TrustID,
+			},
+		}
+	} else {
+		return nil, token3.ErrScopeEmpty
 	}
-  return authMap, nil
+	return map[string]interface{}{"auth": authMap}, nil
```
Comment (codevulture) [inline openstack/identity/v3/extensions/trust/request.go]: Funtion call --> ao.AuthOptions.ToAuthOptionsV3Map(c, nil)  would always a return an authMap of map[sring]interface type with "auth" as key as per: 
https://github.com/rackspace/gophercloud/blob/master/openstack/identity/v3/tokens/requests.go#L42
https://github.com/rackspace/gophercloud/blob/master/openstack/identity/v3/tokens/requests.go#L230
So is there a need for type assertion in L16 and this change?Please let me know if i am missing something? Thanks.

Comment (MatMaul) [inline openstack/identity/v3/extensions/trust/request.go]: What do you mean ? If I remove the type assertion authMap is then of type interface{} after my first line of change, which means that I can't do authMap["scope"] = something a few lines later without triggering a compile error.

Comment (MatMaul) [inline openstack/identity/v3/extensions/trust/request.go]: hum however I think we can safely remove the explicit cast on this line indeed. I'll try and push if it compiles.

Comment (MatMaul) [inline openstack/identity/v3/extensions/trust/request.go]: Ok so it is not an explicit cast but an initialisation, I can't remove it, using {} doesn't initialise a map like in python.

**Comment by codevulture** [comment]
@jrperritt, @MatMaul  : So, i tested this patch and this solves the problem of scope addition as mentioned.Thanks.

**Comment by jrperritt** [comment]
@codevulture OK, thank you for reviewing and testing. +2

**Pull request merged**